Ability to configure the get method

lukemarsh opened the pull request:
I noticed you had a createMethod and an updateMethod so I have added a getMethod. I am using an API which is not particularly RESTful and they only use POST! I have also created a pull request for ng-admin too.

fzaninotto commented on lib/Entity/Entity.js:
```diff
@@ -25,6 +25,7 @@ class Entity {
         this._url = null;
         this._createMethod = null; // manually set the HTTP-method for create operation, defaults to post
         this._updateMethod = null; // manually set the HTTP-method for update operation, defaults to put
+        this._getMethod = null; // manually set the HTTP-method for the get operation, defaults to get
```
Call it `_retrieveMethod` instead, for better compatibility with the CRUD acronym. By the way, it's probably time to add `_deleteMethod`, too.

fzaninotto commented:
Also, please add tests to `EntityTest` and `ReadQueriesTest`

lukemarsh commented:
I have renamed the getMethod and added a deleteMethod. I have added some tests in EntityTest but I couldn't find any tests for passing in custom methods hence why there are no tests there :)

jpetitcolas commented on tests/lib/Entity/EntityTest.js:
```diff
@@ -87,6 +87,30 @@ describe('Entity', function() {
+
+        it('should return null if no deleteMethod has been set', function() {
+            var post = new Entity('post');
+            assert.equal(null, post.deleteMethod());
```
Shouldn't we set `DELETE` as a default?

jpetitcolas commented on tests/lib/Entity/EntityTest.js:
```diff
@@ -87,6 +87,30 @@ describe('Entity', function() {
+
+        it('should return null if no retrieveMethod has been set', function() {
+            var post = new Entity('post');
+            assert.equal(null, post.retrieveMethod());
```
Same

fzaninotto commented:
Thanks!
